tockloader-lib/src/attributes/hardware_attributes.rs

```diff
@@ -0,0 +1,36 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OXIDOS AUTOMOTIVE 2024.
+
+#[derive(Debug)]
+pub struct HardwareAttributes {
+    pub board: Option<String>,
+    pub arch: Option<String>,
+    pub appaddr: Option<String>,
+    pub boothash: Option<String>,
+    pub bootloader_version: Option<String>,
+    pub sentinel: Option<String>,
+    pub kernel_version: Option<u64>,
+    pub app_mem_start: Option<u32>,
+    pub app_mem_len: Option<u32>,
+    pub kernel_bin_start: Option<u32>,
+    pub kernel_bin_len: Option<u32>,
+}
+
```

Comment:
I noticed that `HardwareAttributes` is actually a few different things:
- Kernel attributes (& Kernel version) from `get_kernel_attributes.rs`
- Board Attributes (& some other stuff) from `get_board_attributes.rs`

I would rather have two different structures. Eventually, these two structures alongside the vector of AppAttributes (which, again, could be replaced with `tbf-parser`) be bundled up in a generic `Attributes` struct

```rs
struct Attributes<'a> {
    bootloader: BootloaderAttributes;
    board: BoardAttributes;
    apps: Vec<TbfHeader<'a>>
}
```